apps/api/src/app/messages/usecases/get-messages/get-messages.usecase.ts

```diff
@@ -48,14 +47,35 @@ export class GetMessages {
       }
     }
 
+    const totalCount = await this.messageRepository.count(
+      query,
+      COUNT_LIMIT
+      /*
+       * todo NV-2161 in version 0.16:
+       *  update option as below,
+       *  update below:  hasMore = feed.length < totalCount
+       *  remove totalCount
+       * { skip: command.page * LIMIT, limit: LIMIT + 1 }
+       */
+    );
+
+    const hasMore = this.getHasMore(command.page, LIMIT, data, totalCount);
+
     return {
       page: command.page,
       totalCount,
+      hasMore,
       pageSize: LIMIT,
       data,
     };
   }
 
+  private getHasMore(page: number, LIMIT: number, feed, totalCount) {
```

Comment:
maybe it should be moved to the base repository and reused?

Reply:
I am not sure because this logic will be changed in version 0.16 , after the update, we could rethink it and add the combined query in the repository (for pagination) and the common part in base repository.